File: src/app/objetos/Categoria.ts
export class Categoria{
    _codigo:number;
    _nombre:String;
    
    constructor(nombre:String, codigo:number){
        this._nombre = nombre;
        this._codigo = codigo;
    }

    get nombre(){
        return this.nombre;
    }

    set nombre(nombre:String){
        this._nombre = nombre;
    }
    get codigo(){
        return this._codigo;
    }

    set codigo(codigo:number){
        this._codigo = codigo;
    }
}

File: Backend/IPC2Proyecto2/backendRevistas/src/main/java/Modelo/Categoria.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Modelo;

/**
 *
 * @author ordson
 */
public class Categoria {
    private int codigo;
    private String nombre;

    public Categoria(int codigo, String nombre) {
        this.codigo = codigo;
        this.nombre = nombre;
    }

    public Categoria(String nombre) {
        this.nombre = nombre;
    }

    public Categoria() {
    }

    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    @Override
    public String toString() {
        return "Categoria{" + "codigo=" + codigo + ", nombre=" + nombre + '}';
    }
    
}